File: book-exchange-frontend/types/book.ts
export type BookCondition = "NEW" | "GOOD" | "FAIR" | "POOR";

export interface Book {
  [x: string]: unknown;
  id: string;
  title: string;
  author: string;
  condition: BookCondition;
  is_available: boolean;
  created_at: string | number | Date;
  updated_at: string | number | Date;
  owner_id: string;
  owner:{
    id:string;
    username:string;

  }
}

export interface BookCreate {
  title: string;
  author: string;
  condition: BookCondition;
}

File: app/schemas/book.py
from pydantic import BaseModel
from uuid import UUID
from app.enums import BookConditionEnum
from datetime import datetime

class BookCreate(BaseModel):
    title: str
    author: str
    condition: BookConditionEnum
    # description: str
    

class UserBasic(BaseModel):
    id:UUID
    username:str

    class Config:
        from_attributes = True


class BookResponse(BaseModel):
    id: UUID
    title: str
    author: str
    condition: BookConditionEnum
    # description:str
    owner_id: UUID
    owner:UserBasic
    is_available: bool
    created_at: datetime
    updated_at: datetime
     

    class Config:
        from_attributes = True
